### ISSUE TITLE
Lark problems

### ISSUE THREAD

**Issue by linuxwanted** [INTI-CMNB/KiBot#609]
Fill as much as you can, remove the rest

**Describe the bug**
HTML and XLSX bom breaks with KiCAD 8 and KiBot 1.6.5 (Ubuntu 22.04/installed via pip) on new designs (some old updated designs KiCAD6->7->8 seems to work) Any Idea how I can debug further, no Idea about Lark ...

**To Reproduce**
Command line options passed to KiBot, configuration files (.kibot.yaml), PCB/Schematic example to reproduce it (Ex: a link to a repo where the files are)
kibot -v
bom.kibot.yaml:

```yaml
kibot:
  version: 1

global:
  filters:
  - number: 1007
  - number: 1015
  - number: 58

outputs:

- name: basic_ibom
  comment: Interactive HTML BoM
  type: ibom
  dir: bom
  options:
    dark_mode: true
    highlight_pin1: true

# special logos at:
# /media/syncthing/kicad/kibot/IBRW_Logo.png
# /media/syncthing/kicad/kibot/IBRW_Logo2.png

- name: generic_bom_html
  comment: Generic Bill of Materials in HTML format
  type: bom
  dir: bom
  options:
    format: HTML
    group_fields: &id001
    - part
    - part lib
    - value
    - footprint
    - footprint lib
    - voltage
    - tolerance
    - current
    - power
    - 'mfg#'
    count_smd_tht: true
    #component_aliases: [ ['r'], ['l'], ['c'], ['sw'], ['zener'], ['d'] ]
    conf:
      exclude_any: &id002
        - column: References
          regex: '^TP[0-9]*'
        - column: References
          regex: '^FID'
        - column: Part
          regex: 'mount.*hole'
        - column: Part
          regex: 'solder.*bridge'
        - column: Part
          regex: 'test.*point'
        - column: Footprint
          regex: 'test.*point'
        - column: Footprint
          regex: 'mount.*hole'
        - column: Footprint
          regex: 'fiducial'
        - column: References
          regex: 'LOGO'
    html:
      logo: '/media/syncthing/kicad/kibot/IBRW_Logo.png'
      title: 'Bill of materials'

- name: generic_bom_csv
  comment: Generic Bill of Materials in CSV format
  type: bom
  dir: bom
  options:
    format: CSV
    group_fields: *id001
    conf:
      exclude_any: *id002


- name: generic_bom_xlsx
  comment: Generic Bill of Materials in XLSX format
  type: bom
  dir: bom
  options:
    format: XLSX
    group_fields: *id001
    conf:
      exclude_any: *id002
    count_smd_tht: true
    xlsx:
      logo: '/media/syncthing/kicad/kibot/IBRW_Logo2.png'
      logo_width: 370
      title: 'Bill of materials'

- name: costs_bom_xlsx
  comment: Costs Bill of Materials in XLSX format
  type: bom
  dir: bom
  options:
    format: XLSX
    group_fields: *id001
    conf:
      exclude_any: *id002
    count_smd_tht: true
    output: '%f-%i%I%v-cost.%x'
    xlsx:
      style: modern-green
      kicost: true
      specs: true
      kicost_api_disable: KitSpace
      logo: '/media/syncthing/kicad/kibot/IBRW_Logo2.png'
      logo_width: 370
      title: 'Bill of materials'
    pre_transform: _kicost_rename
```

**Expected behavior**
boms should be generated

**Screenshots**
debug output via kibot -v:

- 'Generic Bill of Materials in HTML format' (generic_bom_html) [bom]
DEBUG:Output destination: /media/nfs/files/projekte/KiCad_explore/SmartESP32-S3/bom (kibot - kiplot.py:443)
DEBUG:Filters reset (kibot - fil_base.py:194)
DEBUG:Applying filter `_mechanical` to exclude (kibot - fil_base.py:187)
DEBUG:Applying filter `_kibom_dnf_Config` to fitted (kibot - fil_base.py:205)
DEBUG:Applying filter `_kibom_dnc_Config` to fixed (kibot - fil_base.py:215)
DEBUG:Applying KiBoM style variants `default` (kibot - var_kibom.py:106)
DEBUG:Creating internal filter: {'type': 'expand_text_vars', 'comment': 'Internal default text variables expander', 'name': '_expand_text_vars'} (kibot - fil_base.py:320)
DEBUG:Applying transform filter `_expand_text_vars` (kibot - fil_base.py:173)
DEBUG:Reading KiCad config from `/home/ralf/.config/kicad/8.0/kicad_common.json` (kibot - config.py:441)
WARNING:(W009) KiCad config without environment.vars section (kibot - config.py:358)
DEBUG:Using KICAD8_SYMBOL_DIR="/usr/share/kicad/symbols" (guessed) (kibot - config.py:416)
DEBUG:Using KICAD8_TEMPLATE_DIR="/usr/share/kicad/template" (guessed) (kibot - config.py:416)
DEBUG:Using KICAD_USER_TEMPLATE_DIR="/home/ralf/.local/share/kicad/8.0/template" (guessed) (kibot - config.py:416)
DEBUG:Using KICAD8_FOOTPRINT_DIR="/usr/share/kicad/footprints" (guessed) (kibot - config.py:416)
DEBUG:Using KICAD8_3DMODEL_DIR="/usr/share/kicad/3dmodels" (guessed) (kibot - config.py:416)
DEBUG:Using KICAD8_3RD_PARTY="/home/ralf/.local/share/kicad/6.0/3rdparty" (guessed) (kibot - config.py:416)
DEBUG:Exporting KIPRJMOD="/media/nfs/files/projekte/KiCad_explore/SmartESP32-S3" (kibot - config.py:462)
DEBUG:Also exporting KICAD6_SYMBOL_DIR="/usr/share/kicad/symbols" (kibot - config.py:470)
DEBUG:Also exporting KICAD7_SYMBOL_DIR="/usr/share/kicad/symbols" (kibot - config.py:470)
DEBUG:Also exporting KICAD6_TEMPLATE_DIR="/usr/share/kicad/template" (kibot - config.py:470)
DEBUG:Also exporting KICAD7_TEMPLATE_DIR="/usr/share/kicad/template" (kibot - config.py:470)
DEBUG:Also exporting KICAD6_FOOTPRINT_DIR="/usr/share/kicad/footprints" (kibot - config.py:470)
DEBUG:Also exporting KICAD7_FOOTPRINT_DIR="/usr/share/kicad/footprints" (kibot - config.py:470)
DEBUG:Also exporting KICAD6_3DMODEL_DIR="/usr/share/kicad/3dmodels" (kibot - config.py:470)
DEBUG:Also exporting KICAD7_3DMODEL_DIR="/usr/share/kicad/3dmodels" (kibot - config.py:470)
DEBUG:Also exporting KICAD6_3RD_PARTY="/home/ralf/.local/share/kicad/6.0/3rdparty" (kibot - config.py:470)
DEBUG:Also exporting KICAD7_3RD_PARTY="/home/ralf/.local/share/kicad/6.0/3rdparty" (kibot - config.py:470)
DEBUG:No 3D aliases (/home/ralf/.config/kicad/8.0/3d/3Dresolver.cfg) (kibot - config.py:578)
DEBUG:Decimal point `,` (kibot - units.py:196)
Traceback (most recent call last):
  File "/home/ralf/.local/bin/kibot", line 8, in <module>
    sys.exit(main())
  File "/home/ralf/.local/lib/python3.10/site-packages/kibot/__main__.py", line 546, in main
    generate_outputs(outputs, args.target, args.invert_sel, args.skip_pre, args.cli_order, args.no_priority,
  File "/home/ralf/.local/lib/python3.10/site-packages/kibot/kiplot.py", line 599, in generate_outputs
    _generate_outputs(outputs, targets, invert, skip_pre, cli_order, no_priority, dont_stop)
  File "/home/ralf/.local/lib/python3.10/site-packages/kibot/kiplot.py", line 589, in _generate_outputs
    run_output(out, dont_stop)
  File "/home/ralf/.local/lib/python3.10/site-packages/kibot/kiplot.py", line 493, in run_output
    out.run(get_output_dir(out.dir, out))
  File "/home/ralf/.local/lib/python3.10/site-packages/kibot/out_base.py", line 214, in run
    self.options.run(target)
  File "/home/ralf/.local/lib/python3.10/site-packages/kibot/out_bom.py", line 987, in run
    do_bom(output, format, comps, self)
  File "/home/ralf/.local/lib/python3.10/site-packages/kibot/bom/bom.py", line 532, in do_bom
    groups = group_components(cfg, comps)
  File "/home/ralf/.local/lib/python3.10/site-packages/kibot/bom/bom.py", line 435, in group_components
    c.value_sort = comp_match(c.value, c.ref_prefix, c.ref, warn_extra=True)
  File "/home/ralf/.local/lib/python3.10/site-packages/kibot/bom/units.py", line 228, in comp_match
    result = parse(ref_prefix[0]+' '+with_commas, with_extra=True, stronger=stronger)
  File "/home/ralf/.local/lib/python3.10/site-packages/kibot/bom/electro_grammar.py", line 205, in parse
    initialize()
  File "/home/ralf/.local/lib/python3.10/site-packages/kibot/bom/electro_grammar.py", line 201, in initialize
    parser = Lark(g, start='main')  # , debug=DEBUG)
  File "/home/ralf/.local/lib/python3.10/site-packages/lark/lark.py", line 207, in __init__
    self.terminals, self.rules, self.ignore_tokens = self.grammar.compile(self.options.start)
  File "/home/ralf/.local/lib/python3.10/site-packages/lark/load_grammar.py", line 491, in compile
    terminals = [TerminalDef(name, transformer.transform( term_tree ), priority)
  File "/home/ralf/.local/lib/python3.10/site-packages/lark/load_grammar.py", line 491, in <listcomp>
    terminals = [TerminalDef(name, transformer.transform( term_tree ), priority)
  File "/home/ralf/.local/lib/python3.10/site-packages/lark/visitors.py", line 138, in transform
    tree = t.transform(tree)
  File "/home/ralf/.local/lib/python3.10/site-packages/lark/visitors.py", line 84, in transform
    return self._transform_tree(tree)
  File "/home/ralf/.local/lib/python3.10/site-packages/lark/visitors.py", line 80, in _transform_tree
    children = list(self._transform_children(tree.children))
  File "/home/ralf/.local/lib/python3.10/site-packages/lark/visitors.py", line 71, in _transform_children
    yield self._transform_tree(c)
  File "/home/ralf/.local/lib/python3.10/site-packages/lark/visitors.py", line 81, in _transform_tree
    return self._call_userfunc(tree, children)
  File "/home/ralf/.local/lib/python3.10/site-packages/lark/visitors.py", line 47, in _call_userfunc
    return f(children)
  File "/home/ralf/.local/lib/python3.10/site-packages/lark/load_grammar.py", line 418, in expansion
    raise GrammarError("Lark doesn't support joining terminals with conflicting flags!"+str(items))
lark.exceptions.GrammarError: Lark doesn't support joining terminals with conflicting flags!['(?i:(?:(?i:(?:(?i:CAPACITOR)|(?i:CAPA)))|(?i:(?i:C)(?i:(?:(?i:AP))?))))', '(?:(?:[ \t\x0c\r\n])+|[,;])']

**Environment (please complete the following information):**
Where are you running KiBot:
- Using a desktop system? Which OS, distro and version?
KiCAD 8 and KiBot 1.6.5 (Ubuntu 22.04 kibot installed via pip3)

**Additional context**

**Comment by set-soft** [comment]
This seems to be some issue with the Lark version you have installed. Try running `kibot-check` to find which version is in use.

Note that the conflict is with some Lark installed at user level (/home/ralf/.local/lib/python3.10/site-packages/lark/load_grammar.py), not an official package. The problem will most probably go away if you uninstall this local copy. The version shipped with Ubuntu 22.04 should work.

**Comment by set-soft** [comment]
> * Using a desktop system? Which OS, distro and version?
>   KiCAD 8 and KiBot 1.6.5 (Ubuntu 22.04 kibot installed via pip3)

BTW: Why are you using pip in on a Debian based system? You have Debian packages [here](https://github.com/set-soft/debian). The pip tool is designed to help Windows users and promote experimental chaos on Linux systems, it only woks when using virtual envs, but if you want to waste disk space just use docker images, they work far better.

**Comment by set-soft** [comment]
Ok, I tried with the last available version of Lark (1.1.9) and all works perfectly.

So I guess you have a very old Lark installed, just upgrade it.
